libs/components/datetime/src/lib/modules/date-range-picker/date-range-picker.component.ts

```diff
@@ -551,13 +545,13 @@ export class SkyDateRangePickerComponent
   ): void {
     const oldValue = this.#getValue();
 
-    const valueOrDefault =
-      !value || isNullOrUndefined(value.calculatorId)
-        ? this.#getDefaultValue(this.calculators[0])
-        : {
-            ...this.#getDefaultValue(this.#getCalculator(value.calculatorId)),
-            ...value,
-          };
+    const isValueEmpty = !value || isNullOrUndefined(value.calculatorId);
+    const valueOrDefault = isValueEmpty
```

Comment:
The logic of this condition didn't change; I just tweaked it a bit for readability.